admin/create-dumps.sh

```diff
@@ -47,8 +52,45 @@ chown "$BACKUP_USER:$BACKUP_GROUP" \
       "$BACKUP_DIR"/fullexport/"$DUMP_NAME"
 echo "Backup directories created!"
 
-echo "Begin copying..."
+# Copy the files into the backup directory
+echo "Begin copying dumps to backup directory..."
 chown "$BACKUP_USER:$BACKUP_GROUP" "$DUMP_DIR"/*
 chmod "$BACKUP_FILE_MODE" "$DUMP_DIR"/*
 retry cp -a "$DUMP_DIR"/* "$BACKUP_DIR"/fullexport/"$DUMP_NAME"/
 echo "Dumps copied to backup directory!"
+
+# rsync the files into the FTP server
+# TODO (param): remove old dumps
+
+FTP_CURRENT_DUMP_DIR="$FTP_DIR"/fullexport/"$DUMP_NAME"
+
+# create the dir in which to copy the dumps before
+# changing their permissions to the FTP_FILE_MODE
+echo "Creating FTP directories and setting permissions..."
+mkdir -m "$FTP_DIR_MODE" -p \
+         "$FTP_DIR"/fullexport/ \
+         "$FTP_CURRENT_DUMP_DIR"
+
+# make sure these dirs are owned by the correct user
+chown "$FTP_USER:$FTP_GROUP" \
+      "$FTP_DIR" \
+      "$FTP_DIR"/fullexport/ \
+      "$FTP_CURRENT_DUMP_DIR"
+
+# make sure all dump files are owned by the correct user
+# and set appropriate mode for files to be uploaded to
+# the FTP server
+chown "$FTP_USER:$FTP_GROUP" "$DUMP_DIR"/*
+chmod "$FTP_FILE_MODE" "$DUMP_DIR"/*
+retry cp -a "$DUMP_DIR"/* "$FTP_CURRENT_DUMP_DIR"
+
+# create an explicit rsync filter for the new private dump in the
+# ftp folder
+touch "$FTP_CURRENT_DUMP_DIR"/.rsync-filter
+PRIVATE_DUMP_RULE=`printf "exclude listenbrainz-private-dump-%s.tar.xz" $DUMP_TIMESTAMP`
```

Comment:
This is OK, but I would really prefer to see an explicit rule to copy the public files to the public FTP, rather than an explicit rule to exclude the private dump. If something with the exclude private rule goes wrong, private data gets exposed. If something goes with with the include public rule, nothing gets copied. I know which error I prefer to have.

Reply:
Very reasonable, makes sense. I'll do it.